roles/grader/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+# tasks file for grader
+- name: Exchange directory
+  ansible.builtin.file:
+    path: '{{ exchange_root }}'
+    state: directory
+    mode: '0777'
+- name: /etc/jupyter
+  ansible.builtin.file:
+    path: /etc/jupyter
+    state: directory
+    mode: '0755'
+- name: Global nbgrader config
+  ansible.builtin.copy:
+    dest: /etc/jupyter/nbgrader_config.py
+    mode: 0644
+    content: |
+      c = get_config()
+      c.Exchange.root = '{{ exchange_root }}'
+- name: Grader jupyter dir
+  ansible.builtin.file:
+    path: /home/{{ grader_user }}/.jupyter
+    state: directory
+    owner: '{{ grader_user }}'
+    group: '{{ grader_user }}'
+    mode: '0755'
+- name: Nbgrader config for grader user
+  ansible.builtin.copy:
+    dest: /home/{{ grader_user }}/.jupyter/nbgrader_config.py
+    mode: 0644
+    owner: '{{ grader_user }}'
+    group: '{{ grader_user }}'
+    content: |
+      c = get_config()
+      c.CourseDirectory.course_id = '{{ course_id }}'
+      c.CourseDirectory.root = '/home/{{ grader_user }}/{{ course_id }}'
+- name: Course directory
+  ansible.builtin.file:
+    path: '/home/{{ grader_user }}/{{ course_id }}'
+    state: directory
+    owner: '{{ grader_user }}'
+    group: '{{ grader_user }}'
+    mode: '0755'
+- name: Clone course repo
+  ansible.builtin.git:
+    repo: '{{ course_repo }}'
+    dest: '/home/{{ grader_user }}/{{ course_id }}/source'
+    version: '{{ course_version }}'
+  become: true
+  become_user: "{{ grader_user }}"
+- name: Create students
+  when: students is defined
+  ansible.builtin.user:
+    name: "{{ item[0] }}"
+    password: "{{ item[1] | password_hash(hashtype='sha512') }}"
+    shell: /bin/bash
+  loop: "{{ students | from_json }}"
+- name: Add students to course
+  when: students is defined
+  ansible.builtin.command: '{{ conda_environment_bin }}/nbgrader db student add "{{ item[0] }}"'
+  environment:
+    PATH: "{{ conda_environment_bin }}:{{ ansible_env.PATH }}"
+  loop: "{{ students | from_json }}"
+  args:
+    chdir: "/home/{{ grader_user }}/{{ course_id }}"
+  become: true
+  become_user: "{{ grader_user }}"
```

Comment:
Are new student accounts that are added e.g. through sram automatically added to the course?

Reply:
This task is runs once during provisioning.

I could add add command to [user sync cron job script](https://gitlab.com/rsc-surf-nl/plugins/plugin-co/-/blob/main/files/cron_user.sh?ref_type=heads#L72) , but that will be very flaky.

So I would rather wait for the lms integration where we will fetch usernames from lms and create them in VM.